src/read.c

```diff
@@ -1401,9 +1401,7 @@ static avifBool avifDecoderGenerateImageGridTiles(avifDecoder * decoder, avifIma
                     decoder->imageCount = tile->input->samples.count;
                 }
             }
-        }
-
-        if (memcmp(item->type, firstTileItem->type, 4)) {
+        } else if (memcmp(item->type, firstTileItem->type, 4)) {
```

Comment:
This change is optional. Please let me know if you'd like me to undo this change.

This mimics the control structure of the original code (before commit 63a496c80343b67e1d5e84f984cc275a2986ee43):

```
                // The coding format is defined by the item type.
                if (firstTile == NULL) {
                    firstTile = tile;
                } else if (memcmp(tile->type, firstTile->type, 4)) {
                    avifDiagnosticsPrintf(diag,
                                          "Tile item ID %u of type '%.4s' differs from other tile type '%.4s'",
                                          tile->id,
                                          (const char *)tile->type,
                                          (const char *)firstTile->type);
                    return AVIF_RESULT_BMFF_PARSE_FAILED;
                }
```